src/dex-ui/services/JsonRpcService/Factory.ts

```diff
@@ -31,8 +55,35 @@ async function getPair(tokenAAddress: string, secondTokenAddress: string, transa
   return await factoryContract.getPair(tokenAAddress, secondTokenAddress, transactionFee);
 }
 
+/**
+ * Fetches the best pair available for swap.
+ * @returns A pair contract addresses.
+ */
+async function getBestSwapPairAvailable(params: GetBestSwapPairAvailableProps): Promise<PairDataResponse | undefined> {
+  const { tokenAAddress, tokenBAddress, tokenAQty } = params;
+  // Since `recommendedPairToSwap` is not accepting bignumber, sending token quantity with hard coded precision value
+  const tokenToTradeAmount = tokenAQty * 100000000;
+
```

Comment:
The SC method `recommendedPairToSwap ` is not accepting `BigNumber` as parameter due to we did multiplied with precision here. which exposes a issue In case user enters a Input with more than 8 numbers after decimal point the SC method gives overflow error